Jint/Runtime/Interpreter/Expressions/JintNewExpression.cs

```diff
@@ -21,14 +22,22 @@ protected override void Initialize()
             for (var i = 0; i < _jintArguments.Length; i++)
             {
                 _jintArguments[i] = Build(_engine, (Expression) expression.Arguments[i]);
+                _hasSpreads |= _jintArguments[i] is JintSpreadExpression;
             }
         }
 
         protected override object EvaluateInternal()
         {
-            var arguments = _engine._jsValueArrayPool.RentArray(_jintArguments.Length);
-
-            BuildArguments(_jintArguments, arguments);
+            JsValue[] arguments;
+            if (_hasSpreads)
+            {
+                arguments = BuildArgumentsWithSpreads(_jintArguments);
+            }
+            else
+            {
+                arguments = _engine._jsValueArrayPool.RentArray(_jintArguments.Length);
```

Comment:
Is there a 'release' operation on these?

Reply:
Yes it's there at the end. Custom build array (from List.ToArray) can also be returned, it's currently put back to pool only if size is in the range of 1...4.